lib/nmatrix/math.rb

```diff
@@ -155,6 +155,57 @@ def getrf!
 
   #
   # call-seq:
+  #     geqrf! -> shape.min x 1 NMatrix 
+  #
+  # QR factorization of a general M-by-N matrix +A+. 
+  # The QR factorization is A = QR, where Q is orthogornal and R is Upper Triangular
+  # +A+ is overwritten with the elements of R and Q with Q being represented by the 
+  # elements below A's diagonal and an array of scalar factors in the output NMatrix. 
+  #  
+  # Only works for dense matrices.
+  #
+  # * *Returns* :
+  #   - Vector TAU. Q(partial) and R are stored in A.
+  # * *Raises* :
+  #   - +StorageTypeError+ -> ATLAS functions only work on dense matrices.
+  #
+  def geqrf!
+    raise(StorageTypeError, "ATLAS functions only work on dense matrices") unless self.dense?
+    
+    temp = self.transpose
+    tau = NMatrix.new([self.shape.min,1], dtype: self.dtype)
+    NMatrix::LAPACK::lapacke_geqrf(:col, self.shape[0], self.shape[1], temp, self.shape[0], tau)
```

Comment:
Nothing in math.rb should directly call a lapacke_* function. All calls to lapacke_* functions should be in lapacke.rb. You can just move `#geqrf!` directly to lapacke.rb, I think.